kong/plugins/jwt/api.lua

```diff
@@ -11,11 +12,35 @@ return {
       crud.paginated_set(self, dao_factory.jwt_secrets)
     end,
 
-    PUT = function(self, dao_factory)
+    PUT = function(self, dao_factory, helpers)
+      if self.params.algorithm == "RS256" and self.params.rsa_public_key == nil then
+          return helpers.responses.send_HTTP_BAD_REQUEST("No mandatory 'rsa_public_key'")
+       end
+
+      if self.params.algorithm == "RS256" and crypto.pkey.from_pem(self.params.rsa_public_key) == nil then
+        return helpers.responses.send_HTTP_BAD_REQUEST("'rsa_public_key' format is invalid")
+      end
+
+      if self.params.algorithm == "HS256" and self.params.secret == nil then
+        return helpers.responses.send_HTTP_BAD_REQUEST("No mandatory 'secret'")
+      end
```

Comment:
The API is not the right place to perform schema validation. Schema validation must be performed at the Model layer, aka in the `schema.lua` file.